src/daft-stats/src/column_stats/mod.rs

```diff
@@ -116,6 +142,17 @@ impl ColumnRangeStatistics {
         let _num_bytes = series.size_bytes().unwrap();
         Self::Loaded(lower, upper)
     }
+
+    /// Casts the internal [`Series`] objects to the specified DataType
+    pub fn cast(&self, dtype: &DataType) -> crate::Result<Self> {
```

Comment:
We shouldn't be able to cast ColumnStats naively like this. I dont think Cast preserves min / max guarantees that ColumnStats needs. also wouldn't this allow be casted to a non `supports_dtype()`

Reply:
Changed to only cast certain allowed types, otherwise fall-back onto `Missing`